File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers, visited, v):
    visited[v] = 1
    
    for i in range(len(computers)):
        if not visited[i] and v != i and computers[v][i] == 1:
            dfs(computers, visited, i)
            
def solution(n, computers):
    answer = 0 # 네트워크 개수
    visited = [0] * (n+1)
    
    for i in range(n):
        if not visited[i]:
            dfs(computers, visited, i)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public void dfs(int[][] computers, int[] visited, int v) {
        visited[v] = 1;
        
        for (int i=0; i<computers.length; i++) {
            if (visited[i] == 0 && computers[v][i] == 1) {
                dfs(computers, visited, i);
            }
        }
    }
    public int solution(int n, int[][] computers) {
        int answer = 0;
        int[] visited = new int[n+1];
        // System.out.println(Arrays.toString(visited));
        
        for (int i=0; i<n; i++) {
            if (visited[i] == 0) {
                dfs(computers, visited, i);
                answer ++;
            }
        }
        return answer;
    }
}